crates/sui-indexer-alt-framework/src/ingestion/streaming_client.rs

```diff
@@ -92,32 +103,80 @@ pub mod test_utils {
             self: Pin<&mut Self>,
             _cx: &mut std::task::Context<'_>,
         ) -> std::task::Poll<Option<Self::Item>> {
-            let mut checkpoints = self.checkpoints.lock().unwrap();
-            if checkpoints.is_empty() {
+            let mut actions = self.actions.lock().unwrap();
+            if actions.is_empty() {
                 return std::task::Poll::Ready(None);
             }
-            let result = checkpoints.remove(0);
-            std::task::Poll::Ready(Some(result.map(|seq| {
-                let mut builder = TestCheckpointBuilder::new(seq);
-                builder.build_checkpoint()
-            })))
+            let action = actions.remove(0);
```

Comment:
It would be easier to implement this by peeking the first action, and then call `actions.remove(0)` only in the cases where you have actually consumed the action, but otherwise, for timeouts, just modify it in place.